src/test/resources/cypher-directive-tests.adoc

```diff
@@ -0,0 +1,262 @@
+----
+
+
+=== render cypher query directive params args
+
+.Request
+[source,graphql]
+----
+query($pname:String) { p3(name:$pname) { id }}
+----
+
+.Request params
+[source,json,request=true]
+----
+{"pname":"foo"}
+----
+
+.Cypher params
+[source,json]
+----
+{"pname":"foo"}
+----
+
+.Cypher
+[source,cypher]
+----
+UNWIND apoc.cypher.runFirstColumnSingle('WITH $name AS name MATCH (p:Person) WHERE p.name = name RETURN p LIMIT 1', { name:$pname }) AS p3 RETURN p3 { .id } AS p3
+----
+
+=== render cypher field directive with params defaults
+
+.Request
+[source,graphql]
+----
+{ person { age }}
+----
+
+.Request params
+[source,json,request=true]
+----
+{}
+----
+
+.Cypher params
+[source,json]
+----
+{"personMult":13}
+----
+
+.Cypher
+[source,cypher]
+----
+MATCH (person:Person) RETURN person { age:apoc.cypher.runFirstColumnMany('WITH $this AS this, $mult AS mult RETURN this.age * mult as age', { this:person, mult:$personMult }) } AS person
+----
+
+=== render cypher query directive
+
+.Request
+[source,graphql]
+----
+{ p2 { id }}
+----
+
+.Request params
+[source,json,request=true]
+----
+{}
+----
+
+.Cypher params
+[source,json]
+----
+{}
+----
+
+.Cypher
+[source,cypher]
+----
+UNWIND apoc.cypher.runFirstColumnMany('MATCH (p:Person) RETURN p', { }) AS p2 RETURN p2 { .id } AS p2
+----
+
+=== render cypher mutation directive
+
+.Request
+[source,graphql]
+----
+mutation { person: createPerson(name:"Joe") { id }}
+----
+
+.Request params
+[source,json,request=true]
+----
+{}
+----
+
+.Cypher params
+[source,json]
+----
+{"personName":"Joe"}
+----
+
+.Cypher
+[source,cypher]
+----
+CALL apoc.cypher.doIt('WITH $name AS name CREATE (p:Person) SET p.name = name RETURN p', { name:$personName }) YIELD value WITH value[head(keys(value))] AS person RETURN person { .id } AS person
+----
+
+=== render cypher field directive with params
+
+.Request
+[source,graphql]
+----
+{ person { age(mult:25) }}
+----
+
+.Request params
+[source,json,request=true]
+----
+{}
+----
+
+.Cypher params
+[source,json]
+----
+{"personMult":25}
+----
+
+.Cypher
+[source,cypher]
+----
+MATCH (person:Person) RETURN person { age:apoc.cypher.runFirstColumnMany('WITH $this AS this, $mult AS mult RETURN this.age * mult as age', { this:person, mult:$personMult }) } AS person
+----
+
+=== render cypher field directive nested
+
+.Request
+[source,graphql]
+----
+{ person { friends { id } }}
+----
+
+.Request params
+[source,json,request=true]
+----
+{}
+----
+
+.Cypher params
+[source,json]
+----
+{}
+----
+
+.Cypher
+[source,cypher]
+----
+MATCH (person:Person)
+RETURN person { friends:[personFriends IN
+  apoc.cypher.runFirstColumnMany('WITH $this AS this  MATCH (this)-[:KNOWS]-(o) RETURN o', { this:person }) | personFriends { .id }] } AS person
+----
+
+=== render cypher query directive params
+
+.Request
+[source,graphql]
+----
+{ p3(name:"Jane") { id }}
+----
+
+.Request params
```

Comment:
I wouldn't call them `Request params` but `Query variables` to stay in the language spec speak.